apps/ejabberd/src/mod_caps.erl

```diff
@@ -0,0 +1,759 @@
+%%%----------------------------------------------------------------------
+%%% File    : mod_caps.erl
+%%% Author  : Magnus Henoch <henoch@dtek.chalmers.se>
+%%% Purpose : Request and cache Entity Capabilities (XEP-0115)
+%%% Created : 7 Oct 2006 by Magnus Henoch <henoch@dtek.chalmers.se>
+%%%
+%%%
+%%% ejabberd, Copyright (C) 2002-2015   ProcessOne
+%%%
+%%% This program is free software; you can redistribute it and/or
+%%% modify it under the terms of the GNU General Public License as
+%%% published by the Free Software Foundation; either version 2 of the
+%%% License, or (at your option) any later version.
+%%%
+%%% This program is distributed in the hope that it will be useful,
+%%% but WITHOUT ANY WARRANTY; without even the implied warranty of
+%%% MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+%%% General Public License for more details.
+%%%
+%%% You should have received a copy of the GNU General Public License along
+%%% with this program; if not, write to the Free Software Foundation, Inc.,
+%%% 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+%%%
+%%% 2009, improvements from ProcessOne to support correct PEP handling
+%%% through s2s, use less memory, and speedup global caps handling
+%%%----------------------------------------------------------------------
+
+-module(mod_caps).
+
+-author('henoch@dtek.chalmers.se').
+
+-protocol({xep, 115, '1.5'}).
```

Comment:
To be consistent with other our modules we should use `-xep` here instead of `-protocol`